omnigibson/transition_rules.py

```diff
@@ -615,39 +743,35 @@ def transition(self, individual_objects, group_objects):
                 t_results.remove.append(obj)
 
         # Remove all physical particles that are inside the blender
-        for system_name in self.particle_requirements.keys():
+        for system_name in cls.particle_requirements.keys():
             system = get_system(system_name)
             # No need to check for whether particle instancers exist because they must due to @self.condition passing!
             for inst in system.particle_instancers.values():
-                indices = self._check_in_volume[blender.name](inst.particle_positions).nonzero()[0]
+                indices = cls._CHECK_IN_VOLUME[blender.name](inst.particle_positions).nonzero()[0]
                 inst.remove_particles(idxs=indices)
 
         # Spawn in blended physical particles!
-        blender.states[Filled].set_value(get_system(self.output_system), True)
+        blender.states[Filled].set_value(get_system(cls.output_system), True)
 
         return t_results
 
-
-"""See the following example for writing simple rules.
-
-  GenericTransitionRule(
-      filters=(
-          AndFilter(CategoryFilter("light"), StateFilter(ToggledOn, True)),
-          CategoryFilter("key"),
-      ),
-      condition_fn=lambda light, key: dist(light, key) < 1,
-      transition_fn=lambda light, key: light.states[ToggledOn].set_value(True),
-  )
-"""
-# TODO: To prevent bugs for now, we make this READ-ONLY (tuple). In the future, rules should be allowed to be added
-# dynamically
-DEFAULT_RULES = (
-    SlicingRule(),
-    DicingRule(),
-    # Strawberry smoothie
-    BlenderRule(
-        output_system="strawberry_smoothie",
-        particle_requirements={"milk": 10},
-        obj_requirements={"strawberry": 5, "ice_cube": 5},
-    ),
+    @classmethod
+    def clear(cls):
+        # Clear blender volume checkers
+        cls._CHECK_IN_VOLUME = dict()
+
+    @classproperty
+    def _do_not_register_classes(cls):
+        # Don't register this class since it's an abstract template
+        classes = super()._do_not_register_classes
+        classes.add("BlenderRuleTemplate")
+        return classes
+
+
+# Create strawberry smoothie blender rule
+StrawberrySmoothieRule = BlenderRuleTemplate.create(
```

Comment:
Aside from the `BlenderRuleTemplate`, we also need to add a `magic stirring rod template` maybe? Or maybe these two can be combined because much logic is shared? Happy to discuss more to flesh out.

Reply:
this is a good point! i think this is analogous to `ToggledOn` requirement, right? there might be a way to unify. let me think about this